src/main/java/org/jenkinsci/plugins/casc/integrations/globalmatrixauth/GroupPermissionDefinition.java

```diff
@@ -0,0 +1,58 @@
+package org.jenkinsci.plugins.casc.integrations.globalmatrixauth;
+
+import hudson.security.Permission;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.util.*;
+import java.util.logging.Logger;
+
+/**
+ * Created by mads on 2/7/18.
+ */
+@Restricted(NoExternalUse.class)
+public class GroupPermissionDefinition {
+
+    private String name;
+    private Collection<String> permissions;
+
+    private static final Logger LOGGER = Logger.getLogger(GroupPermissionDefinition.class.getName());
+
+    @DataBoundConstructor
+    public GroupPermissionDefinition(String name, Collection<String> permissions) {
+        this.name = name;
+        this.permissions = permissions != null ? Collections.unmodifiableCollection(permissions) : Collections.EMPTY_SET;
```

Comment:
`Collections.emptySet()` is typesafely-casted version (will not be hilighted in IDE)